hack/shared-cluster.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# shared-cluster.sh is used to provide a repeatable production cluster create script
+
+LOCATION="westcentralus"
+CLUSTER="shared-cluster"
+RESOURCEGROUP="shared-cluster"
+CLUSTERRESOURCEGROUP="aro-shared-cluster"
+
+usage() {
+	echo -e "usage: ${0} <create|delete>"
+	exit 1
+}
+
+if [ "$#" -ne 1 ]; then
+	usage
+fi
+
+if [[ $1 == create ]]; then
+    echo "creating resource group and network"
+    az group create --name $RESOURCEGROUP --location $LOCATION
+
+    az network vnet create \
+      --resource-group $RESOURCEGROUP \
+      --name aro-vnet \
+      --address-prefixes 10.0.0.0/22
+
+    az network vnet subnet create \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --name master-subnet \
+      --address-prefixes 10.0.0.0/23 \
+      --service-endpoints Microsoft.ContainerRegistry
+
+    az network vnet subnet create \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --name worker-subnet \
+      --address-prefixes 10.0.2.0/23 \
+      --service-endpoints Microsoft.ContainerRegistry
+
+    az network vnet subnet update \
+      --name master-subnet \
+      --resource-group $RESOURCEGROUP \
+      --vnet-name aro-vnet \
+      --disable-private-link-service-network-policies true
+
+    echo "resource group and vnet/subnets created..."
+    echo "creating cluster..."
+    az aro create \
+      --resource-group $RESOURCEGROUP \
+      --cluster-resource-group $CLUSTERRESOURCEGROUP \
+      --name $CLUSTER \
+      --vnet aro-vnet \
+      --master-subnet master-subnet \
+      --worker-subnet worker-subnet || exit 1
+    echo "cluster create completed..."
+    exit 0
+fi
+
+if [[ $1 == "delete" ]]; then
+  az aro delete --name $CLUSTER -g $RESOURCEGROUP --yes
+  az network vnet delete --name aro-vnet -g $RESOURCEGROUP
+  az group delete --resource-group $RESOURCEGROUP --yes
+  exit 0
+fi
```

Comment:
Just a question: should we check for errors before
```bash
exit 0
```
and do
```bash
exit 1
````
in case any (as it's done in the previous if block)? 

(Maybe I am missing something 🙂)

Reply:
I don't want to add complexity/reduce readability into this part of the script and I don't really care if these commands fail. I would actually want the commands to continue if the previous one fails because we just want to delete all of these resources.

As some compromise, I've changed the conditional flow so that the exit is implicit instead of explicit. I required the exit before because there was a function call to `usage` that default ran, and now we avoid the func call of`usage()` through an else statement.